label_studio/core/settings/base.py

```diff
@@ -714,3 +714,7 @@ def collect_versions_dummy(**kwargs):
     CSP_INCLUDE_NONCE_IN = ['script-src', 'default-src']
 
     MIDDLEWARE.append('core.middleware.HumanSignalCspMiddleware')
+
+CLOUD_MAX_CHECKED_OBJECTS = get_env('CLOUD_MAX_CHECKED_OBJECTS', 1000)
```

Comment:
As we discussed in SU, I don't think limiting our search to the first 1k records is a wide enough use case here, unfortunately. I haven't had the chance to look into the alternatives, but I think there are many cases where users need to be able to filter out many files from their bucket using the pattern they provide.

Reply:
Maybe we can add a warning instead of validation error?
We can show such message: "First 1000 object do not match glob pattern. Narrow your glob pattern or proceed if glob pattern is correct"